modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/endpoint/EndpointMatcher.java

```diff
@@ -27,7 +28,8 @@ public EndpointMatcher(List<APIApplication.Endpoint> endpoints) {
 		for (APIApplication.Endpoint endpoint : endpoints) {
 			if (Objects.equals(
 					endpoint.getRetrieveType(),
-					APIApplication.Endpoint.RetrieveType.SINGLE_ELEMENT)) {
+					APIApplication.Endpoint.RetrieveType.SINGLE_ELEMENT) &&
+				Validator.isNotNull(endpoint.getPathParameter())) {
```

Comment:
Instead of checking the path parameter, would it be more obvious to check if method is GET?

Reply:
I also thought about it, but I think this can be more future-proof when we start supporting path parameters in POST endpoints or new Http methods with similar structures get added

Reply:
you are right.